fe/fe-core/src/main/java/org/apache/doris/nereids/analyzer/UnboundRelation.java

```diff
@@ -43,33 +43,41 @@
 public class UnboundRelation extends LogicalLeaf implements Relation, Unbound {
     private final List<String> nameParts;
 
+    private boolean policyChecked;
+
     public UnboundRelation(List<String> nameParts) {
-        this(nameParts, Optional.empty(), Optional.empty());
+        this(nameParts, false);
+    }
+
+    public UnboundRelation(List<String> nameParts, boolean policyChecked) {
+        this(nameParts, policyChecked, Optional.empty(), Optional.empty());
     }
```

Comment:
The policyChecked hacking the UnboundRelation, I can imagine that there will be lots of field need added.
How about generate a CheckPolicy in the LogicalPlanBuilder?
And then you can write a rule to generate a new filter to the logicalRelation, like this:

```java
logicalCheckPolicy(logicalRelation()).thenApply(ctx ->
    LogicalCheckPolicy checkPolicy = ctx.root;
    LogicalRelation relation = checkPolicy.child();
    Optional<String> filter = checkPolicy.getFilter(relation);
    if (filter.isEmpty()) {
        return relation; // no any policy, so we remove the LogicalCheckPolicy Plan
    }
    // has policy, we should remove LogicalCheckPolicy and add a filter to relation
    Expression filter = new NereidsParser().parseExpression(filter.get());
    return new LogicalFilter(filter, relation);
)
```

Reply:
Use logicalRelation can let you not need to care whether the table is exists.

Reply:
So good an idea. I image that the sql of filter is obtained from wherePredicate.toSql() in RowPolicy. But I have difficulty in parsing it, such as "**k[#1] = 1**", due to function toSqlImpl of SlotRef。So I have to use NereidsParser to generate an additional expression of filter when creating a policy. Any good idea?

Reply:
The legacy `Expr` is difficult to convert to Nereids's `Expression`.
Use NereidsParser to parse the originStmt in `RowPolicy` and get the `Expression` is more reasonable.
It is same as parse view sql in each sql which using the view, so you can keep easy.